src/extensions/surveys.tsx

```diff
@@ -483,6 +485,70 @@ export function SurveyPopup({
     const shouldShowConfirmation = isSurveySent || previewPageIndex === survey.questions.length
     const confirmationBoxLeftStyle = style?.left && isNumber(style?.left) ? { left: style.left - 40 } : {}
 
+    // Add URL change listener to hide survey when URL no longer matches
+    useEffect(() => {
+        if (isPreviewMode || !survey.conditions?.url || !posthog) {
+            return
+        }
+
+        const checkUrlMatch = () => {
+            const urlMatchType = survey.conditions?.urlMatchType ?? 'icontains'
+            const url = survey.conditions?.url
+            if (!url) {
+                return
+            }
+            const urlCheck = surveyUrlValidationMap[urlMatchType](url)
+            if (!urlCheck) {
+                setIsPopupVisible(false)
+                removeSurveyFromFocus(survey.id)
+            }
+        }
+
+        // Listen for navigation events
+        const handleUrlChange = () => {
+            checkUrlMatch()
+        }
+
+        // Listen for browser back/forward
+        window.addEventListener('popstate', handleUrlChange)
+
+        // Create our before_send handler
+        const surveyBeforeSend = (eventData: CaptureResult | null) => {
+            if (eventData?.event === '$pageview' || eventData?.event === '$pageleave') {
+                handleUrlChange()
+            }
+            return eventData
+        }
+
+        // Add our handler to the existing before_send configuration
+        const originalBeforeSend = posthog.config.before_send
+        if (isArray(originalBeforeSend)) {
+            posthog.config.before_send = [...originalBeforeSend, surveyBeforeSend]
+        } else if (originalBeforeSend) {
+            posthog.config.before_send = [originalBeforeSend, surveyBeforeSend]
+        } else {
+            posthog.config.before_send = surveyBeforeSend
+        }
```

Comment:
I decided to use the existing PostHog capture events for `$pageview` as they work more reliably across different kinds of apps (SPA, SSR, etc). Let me know if that's okay